src/main/java/org/zalando/nakadi/controller/SubscriptionController.java

```diff
@@ -132,37 +136,42 @@ public SubscriptionController(final SubscriptionDbRepository subscriptionReposit
     }
 
     private ResponseEntity<?> createSubscription(final SubscriptionBase subscriptionBase,
-                                                 final NativeWebRequest request)
+                                                 final NativeWebRequest request,
+                                                 final Client client)
             throws InternalNakadiException, DuplicatedSubscriptionException, ServiceUnavailableException {
-        final List<String> noneExistingEventTypes = checkExistingEventTypes(subscriptionBase);
-        if (!noneExistingEventTypes.isEmpty()) {
-            final String errorMessage = createErrorMessage(noneExistingEventTypes);
-            LOG.debug(errorMessage);
+        final List<EventType> existingEventTypes = getExistingEventTypes(subscriptionBase);
```

Comment:
For me all these methods makes no sence, because the idea of it is pretty strightforward - load everything that is in database, if something is missing - show error. 
```
        final Map<String, Optional<EventType>> eventTypeMapping =
                subscriptionBase.getEventTypes().stream()
                .collect(Collectors.toMap(Function.identity(), t -> eventTypeRepository.findByNameO(t)));
        if (eventTypeMapping.values().stream().filter(o -> !o.isPresent()).findAny().isPresent()) {
            return create(
                    UNPROCESSABLE_ENTITY, 
                    createErrorMessage(eventTypeMapping.entrySet().stream().filter(e -> !e.getValue().isPresent()).map(Map.Entry::getKey).collect(Collectors.toList())), 
                    request);
        }
        eventTypeMapping.values().stream().map(Optional::get).forEach(eventType -> client.checkScopes(eventType.getReadScopes()));

        // generate subscription id and try to create subscription in DB
        final Subscription subscription = subscriptionRepository.createSubscription(subscriptionBase);
        return status(HttpStatus.CREATED).body(subscription);
```

The only problem that I see is that method returning optional throws Exception (findByNameO), It should be replaced with ExecutionException, that wraps InternalNakadiException.